cartographer/io/occupancy_grid.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_IO_OCCUPANCY_GRID_H_
+#define CARTOGRAPHER_IO_OCCUPANCY_GRID_H_
+
+#include "Eigen/Geometry"
+#include "cairo/cairo.h"
+#include "cartographer/io/image.h"
+#include "cartographer/mapping/id.h"
+#include "cartographer/transform/rigid_transform.h"
+
+namespace cartographer {
+namespace io {
+
+constexpr cairo_format_t kCairoFormat = CAIRO_FORMAT_ARGB32;
+
+struct OccupancyGridState {
+  OccupancyGridState(::cartographer::io::UniqueCairoSurfacePtr surface,
+                     Eigen::Array2f origin, Eigen::Array2i size)
+      : surface(std::move(surface)), origin(origin), size(size) {}
+  ::cartographer::io::UniqueCairoSurfacePtr surface;
+  Eigen::Array2f origin;
+  Eigen::Array2i size;
+};
+
+struct SubmapState {
```

Comment:
We call an image of a submap a Slice now in our Ogre code. We should adopt this here as well. How about SubmapSlice?